apps/api/src/domain/models/transcribed-audio/commands/create-transcript.command.integration.spec.ts

```diff
@@ -0,0 +1,222 @@
+            await assertCreateCommandSuccess(assertionHelperDependencies, {
+                systemUserId: dummyAdminUserId,
+                buildValidCommandFSA,
+                initialState: validInitialState,
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id },
+                }: CreateTranscript) => {
+                    const idStatus = await idManager.status(id);
+
+                    expect(idStatus).toBe(NotAvailable);
+
+                    const transcriptSearchResult = await testRepositoryProvider
+                        .forResource<Transcript>(ResourceType.transcribedAudio)
+                        .fetchById(id);
+
+                    expect(transcriptSearchResult).not.toBe(NotFound);
+
+                    expect(transcriptSearchResult).not.toBeInstanceOf(InternalError);
+
+                    const transcript = transcriptSearchResult as unknown as Transcript;
+
+                    assertEventRecordPersisted(transcript, 'TRANSCRIPT_CREATED', dummyAdminUserId);
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the command payload type is invalid', () => {
+            generateCommandFuzzTestCases(CreateTranscript).forEach(
+                ({ description, propertyName, invalidValue }) => {
+                    describe(`when the property ${propertyName} has the invalid value: ${invalidValue} (${description})`, () => {
+                        it('should fail with the appropriate error', async () => {
+                            await assertCommandFailsDueToTypeError(
+                                assertionHelperDependencies,
+                                { propertyName, invalidValue },
+                                buildValidCommandFSA('unused-id')
+                            );
+                        });
+                    });
+                }
+            );
+        });
+
+        describe('when there is no media item with the given `mediaItemId`', () => {
+            it('should fail with the appropriate error', async () => {
+                await assertCreateCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: buildInvalidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat(),
+                    systemUserId: dummyAdminUserId,
+                    checkError: (error) => {
+                        expect(error).toBeInstanceOf(CommandExecutionError);
+
+                        const innerError = error.innerErrors[0];
+
+                        expect(innerError).toBeInstanceOf(InvalidExternalReferenceByAggregateError);
+                    },
+                });
+            });
+        });
+
+        describe('when the MIME type of the media item is not a video or audio format', () => {
+            const disallowedMIMETypes = Object.values(MIMEType).filter(
```

Comment:
Will this be a helper at some point?

Reply:
No. We don't want to couple the implementation to the test. We should repeat the list in the test to be intentional about change.